gravitational/webapps
# Add BannerList and Banner components to display cluster alerts on load

When loading the UI, render cluster alerts.

Currently the only supported alert is for an upgrade notification but this component should handle rendering any alerts with multiple severity levels without updates.

<img width="1399" alt="Screen Shot 2022-09-06 at 11 59 56 AM" src="https://user-images.githubusercontent.com/532033/188706679-46cd636f-bd1f-44d4-b3f7-1712cabfb20a.png">

## QA Notes

When dismissing a banner in the Discover UI it'll appear again in the Main app. This will be resolved in a follow-up branch where banner dismiss persistence will be added.

This code is non functional without the corresponding teleport updates: 
- [teleport pr](https://github.com/gravitational/teleport/pull/15774) 
- and a subsequent PR built on it which adds the apiserver endpoints.

Use [this teleport branch](https://github.com/gravitational/teleport/tree/hatched/cluster-alerts-apiserver) which contains all of the necessary updates and hacks to QA this branch.

Start it with `TELEPORT_UNSTABLE_VC_SYNC_ON_START=yes ./build/teleport start`
